Description: add support for the new tidy-html5 library
Author: Laurent Declercq <l.declercq@nuxwin.com>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
---
 ext/tidy/config.m4 |    8 ++++++++
 ext/tidy/tidy.c    |    5 +++++
 2 files changed, 13 insertions(+)

--- a/ext/tidy/config.m4
+++ b/ext/tidy/config.m4
@@ -25,6 +25,14 @@ if test "$PHP_TIDY" != "no"; then
 
   if test -z "$TIDY_DIR"; then
     AC_MSG_ERROR(Cannot find libtidy)
+  else
+    dnl Check for tidybuffio.h (as opposed to simply buffio.h)
+    dnl which indicates that we are building against tidy-html5
+    dnl and not the legacy htmltidy. The two are compatible,
+    dnl except for with regard to this header file.
+    if test -f "$TIDY_INCDIR/tidybuffio.h"; then
+      AC_DEFINE(HAVE_TIDYBUFFIO_H,1,[defined if tidybuffio.h exists])
+    fi
   fi
 
   TIDY_LIBDIR=$TIDY_DIR/$PHP_LIBDIR
--- a/ext/tidy/tidy.c
+++ b/ext/tidy/tidy.c
@@ -32,7 +32,12 @@
 #include "safe_mode.h"
 
 #include "tidy.h"
+
+#if HAVE_TIDYBUFFIO_H
+#include "tidybuffio.h"
+#else
 #include "buffio.h"
+#endif
 
 /* compatibility with older versions of libtidy */
 #ifndef TIDY_CALL
